
## Opening and Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

<center>
![img](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png)
Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)
</center>

Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of data instead. To avoid spending resources with collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our purpose:

- [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
- [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

Let's start by opening the two data sets and then continue with exploring the data.

# Profitable App profiles foe the app store and google play markets 
Our aim in this project is to find mobile app prodiles that are profitanle for the app store and google play markests. We're working as data analysts for a company that builds Android and IOS mobile apps, and our job is to enable our team of developers to male data-driven decision with respect to the kind od apps they build.

At our company , we only build apps that are free to download and install, and our main source if revenus consists of in-app ads.This means that our revenus for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analusze data to help our developers understand what kind s of apps are likely toattracr more users.


## Opening and Exploring the Data

As of september 2018,there were approximately 2 million Ios apps abailable on the App store, and 2.1 million Android apps on Google play.

![Image](https://raw.githubusercontent.com/lijianqiao-UCL-scua/dataquest_project/main/project%202%20Profitable%20App%20Profiles%20for%20the%20App%20Store%20and%20Google%20Play%20Markets/py1m8_statista.png)
Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)

- [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
- [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

Let's start by opening the two data sets and then continue with exploring the data.

To make it easier to explore the two data sets, we'll first write a function named `explore_data()` that we can use repeatedly to explore rows in a more readable way. We'll also add an option for our function to show the number of rows and columns for any data set.

In [1]:
def explore_data(dataset,start,end,rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')# adds a new(empty) line after each row
    if rows_and_columns:
        print("number of rows",len(dataset))
        print("number of columns",len(dataset[0]))

In [2]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv',encoding='utf-8')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv',encoding='utf-8')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [3]:
opened_file_apple=open('AppleStore.csv',encoding='utf-8')
opened_file_google=open('googleplaystore.csv',encoding='utf-8')

from csv import reader

read_file_apple=reader(opened_file_apple)
read_file_google=reader(opened_file_google)

apple_apps = list(read_file_apple)
ios_header = apple_apps[0]
ios = apple_apps[1:]
google_apps=list(read_file_google)
android_header = google_apps[0]
android = google_apps[1:]

In [4]:
apple_data_demo=explore_data(apple_apps,0,4,rows_and_columns=True)
google_data_demo=explore_data(google_apps,0,4,rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


number of rows 7198
number of columns 16
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'Jan

we have 7198 IOS apps in this data, and the columns that seems intereting are: `'track_name'`, `'currency'`, `'price'`, `'rating_count_tot'`, `'rating_count_ver'`, and `'prime_genre'`. Not all the columns name are self-explanatory in this case , but details about each columns can be found in data set[documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

## delete the Wrong data
The google Play data set has a dedicated [discuss section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), and we can see thay [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) outlines an error for row 10472. Let's print this row and compare it against the header and another row that is correct.

In [5]:
for row in android[1:]:
    if len(row) != len(android_header):
        print(android.index(row))
        print(len(row))

10472
12


In [6]:
print(android_header)
print('\n')
print(len(android_header))
print(android[10472])
print(len(android[10472]))
print(android[1])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


13
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
12
['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


The row 10472 corresponds to the app _Life Made WI-Fi Touchscreen Photo Frame_, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5 (as mentioned in the [discussions section](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015), this problem is caused by a missing value in the `'Category'` column). As a consequence, we'll delete this row. 

In [7]:
print(len(android))
del android[10472]
print(len(android))

10841
10840


## Removing Duplicate Entries

### Part One
If we explore the Google Play data set long enough, we'll find that some apps have more than one entry. For instance, the application Instagram has four entries:

## Removing Duplicate Entries

### Part one 
If we explore the google play data set long enough, we'll find that some apps have more than one entry. For instance, tha application Instagram has four entries:

In [8]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [9]:
duplicate_apps = []
unique_apps=[]
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate:',len(duplicate_apps))
print("\n")
print('Number of duplicate:',duplicate_apps[:15])


Number of duplicate: 1181


Number of duplicate: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


we don't want to count certain apps more than once when we analyza data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows we printed two cells above for the Instagram app, the main difference happends on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was cillected at different times. We can use this to build a criterion for keeping rows. We won't remove rows ramdomly, but rather we'll keep the rows that have the highest number of reviews because the the higher the number of reviews, the more reliable the ratings.

to do that,we will

- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
- Use the dictionary to create a new data set, which will have only one entry per app(and we only select the apps with the highest number of reviews) 

### Part Two

Let's start by building the dictionary.

In [10]:
print('expected length:',len(android)-1181)

expected length: 9659


In a previous code cell, we found that there are 1,181 cases where an app occurs more than once, so the length of our dictionary (of unique apps) should be equal to the difference between the length of our data set and 1,181.

In [11]:
review_max={}
for row in android:
    name = row[0]
    n_review = float(row[3])
    if name in review_max:
        if review_max[name]<n_review:
            review_max[name]=n_review
    else:
        review_max[name]=n_review 

In [12]:
len(review_max)

9659

In [13]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (review_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name) # make sure this is inside the if block

In [14]:
android_clean=[]
already_added=[]
for row in android:
    name=row[0]
    n_review = float(row[3])
    if (review_max[name]==n_review) and (name not in already_added):
            android_clean.append(row) 
            already_added.append(name)


In [15]:
len(android_clean)

9659

## Removing Non-English Apps
### part one

if you expore the data sets enough , you will find the names of the app suggest that they are not design for the enlish use. Below, we see a couple of examples from bote data sets:

In [16]:
def english_detect(string='string'):
    for i in range(len(string)):
        if ord(string[i])>127:
            return False
    return True


In [17]:
list = ['Instagram','爱奇艺PPS -《欢乐颂2》电视剧热播','Docs To Go™ Free Office Suite','Instachat 😜']

for str in list:
    x = english_detect(string=str)
    print(x)

True
False
False
False


In [18]:
def english_detect(string='string'):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True


In [19]:
print(english_detect('Docs To Go™ Free Office Suite'))
print(english_detect('Instachat 😜'))
print(english_detect('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


In [20]:
len(android_clean)

9659

In [21]:
len(ios)

7197

In [22]:
android_english=[]
ios_english=[]
for row in android_clean:
    name = row[0]
    if english_detect(name):
        android_english.append(row)

for row in ios:
    name = row[1]
    if english_detect(name):
        ios_english.append(row)


print(len(android_english))
print(len(ios_english))

9614
6183


We can see that we're left with 9614 android apps and 6183 ios apps.

## Ioslating the Free apps
As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data set contain both free and non-free apps and, we'll need to isolate only the free apps for our analysis. Below, we ios late the free apps for both data sets.

In [23]:
android_final = []
ios_final = []
for row in android_english:
    price = row[7]
    if price == '0':
        android_final.append(row)
for row in ios_english:
    price = row[4]
    if price == '0.0':
        ios_final.append(row)

print(len(android_final))
print(len(ios_final))

8864
3222


We're left with 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.

# Most common apps by Genre
## part one 
As we mentioned in the introduction, our aim is to determine the kins of apps that are likely to attract more user because our revenue is highly influenecd by the number of people using our apps.

To minimize risks and overhead,our validation strategy for an app idea is compriesd of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from user, we then develop it further.
3. If the app is profitble after six months, we also build an IOS version of the app and add it to the App store.


Because our end gooal is to add the app on both the app store and google play, we need to fina app profiles thay are successful om both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market.For this , we'll build a frequency table for the 'prime_genre' colums of the app store 
data set, and the genres and category columns of the google play data set.

### part two 
we'll build two function we can use to analyze the frequency tables

- one function to generate frequency tables that show percentages
- Another function that we can use to display the in a descending order

In [24]:
def freq_table(dataset, index):
    table={}
    total=0
    for row in dataset:
        total+=1
        value = row[index]
        if value in table:
            table[value]+=1
        else:
            table[value]=1
    table_percentages = {}
    for key in table:
        percentage = (table[key]/total) * 100
        table_percentages[key] = percentage
    
    return table_percentages

def display_table(dataset,index):
    table = freq_table(dataset,index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key],key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display,reverse=True)
    for entry in table_sorted:
        print(entry[1],':',entry[0])
        
    

## part three 
We start by examining the frequency table fot the prime_genre columns of the app store dataset

In [25]:
display_table(ios_final,11)


Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among the free English apps, more thayn a half are games.Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed bu social networking apps which amount for 3.29% of the apps in our dataset.

The general impression is that App store(at the lesat the part containing free english apps) is dominated by the apps that are design for the fun (games,enterment, photo and video, social networking , sports ,music,ect.),while apps with practical purpose are more rare.However, the fat that fun apps are the most numberous does't also imply they have the greatest number of user _the demand mighr not be same as the offer.

Let's contunue by examming the Genres and Category columns of the Google play data set.

In [26]:
display_table(android_final,1)# category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

the landscape seems significatly different on google play:there are not that many apps designed for fun, and it seems that a good number of apps are design for practical purpose (family,tools).However , if we investiaget this further we can see thay family category (which accounts for almost 19% of the apps ) means mostly games for kids.

[![img](https://s3.amazonaws.com/dq-content/350/py1m8_family.png)](https://play.google.com/store/apps/category/FAMILY?hl=en)


Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the `Genres` column:

In [27]:
display_table(android_final,-4)
    

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The different between the Genres and the Category columns is not crystal clea, but one things we can notice is that the Genres columns is much more granular (it has more categorues).We're only looking fot the bigger picture at the monment, so we'll only woek with the category columns moving forward.

Up to this point, we found thay the app store is dominated by apps designed for fun,while Google play shows a more balaned landscape of both pratical and for fun apps.Now we'd like to get an idea about the kind of apps that have most users.

## Most Popular apps by genre on the app store 
One way to find out what genres are the most popular(have the most user) is to calculate the average number of installs for each app genre. For google play data set, we can find this information in the install column, but for the app store data set thos imformation is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the acerage number of user ratings per app genre on the app store:

In [28]:
genres_ios = freq_table(ios_final, -5)


In [29]:
for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app =  app[-5]
        if genre_app == genre:
            n_rating=float(app[5])
            total+=n_rating
            len_genre+=1
    average_n_ratings = total/len_genre
    print(genre,':',average_n_ratings)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [30]:
for app in ios_final:
    if app[-5]=='Navigation':
        print(app[1],':',app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [31]:
for app in ios_final:
    if app[-5]=='Games':
        print(app[1],':',app[5])

Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO : 257627
CSR Racing 2 : 257100
Star Wars™: 

The same pattern applies to socal  networking apps, where the average number is heavily influenced by the a few giants like Facebook, Pinterest,skype,ect.Same applies to musica apps ,where a few big players like Pandor,spotify and shazam heavily influence the average number.

Our aim is to find the popular genres, but navigation,social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user rating, while the other apps may struggle ti get past the 10,000
threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the average , but we'll leave this level of detail for the later 

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

However , this niches seems to show some poteneual. One thing we could do is take another popular book and turn it into an app where we could add dofferent features besides the raw version of the book. THis might include daily quotes from the book, an audio version of the book,quizzes about the book,etc. On top of that , we could also embed a dictionary with in the app, so users don't need to exit out app to look up words om an external app.

This idea seems to fit well with the fact that the App stores is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps which means a pratical app might have more of a chance to stand our among the huge number of apps on the App store.

Other genres that seem popular include weather,book, food and drink,or finance.The book genre seem to overlap a bit with the app with the app idea we described above , but the other genres don't seem too interesting to us:


- Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

- Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

- Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

## Most popular apps by genre on google play
For the google play market , we acyually have the data about the number of installs ,s we should be able to get a clearer picture  genre popularity.Howevere, the install number don't seem preceise enough——we can sewe that most values are open-ended(100+,1000+,5000+,etc):


In [32]:
android_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [33]:
display_table(android_final, 5) # the Installs columns

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to `float` — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [39]:
categories_android = freq_table(android_final, 1)
a = {}
for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)
    a[category] = avg_n_installs

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

In [47]:
    table_sorted = sorted(a.items(),reverse=True)
    table_sorted

[('WEATHER', 5074486.197183099),
 ('VIDEO_PLAYERS', 24727872.452830188),
 ('TRAVEL_AND_LOCAL', 13984077.710144928),
 ('TOOLS', 10801391.298666667),
 ('SPORTS', 3638640.1428571427),
 ('SOCIAL', 23253652.127118643),
 ('SHOPPING', 7036877.311557789),
 ('PRODUCTIVITY', 16787331.344927534),
 ('PHOTOGRAPHY', 17840110.40229885),
 ('PERSONALIZATION', 5201482.6122448975),
 ('PARENTING', 542603.6206896552),
 ('NEWS_AND_MAGAZINES', 9549178.467741935),
 ('MEDICAL', 120550.61980830671),
 ('MAPS_AND_NAVIGATION', 4056941.7741935486),
 ('LIFESTYLE', 1437816.2687861272),
 ('LIBRARIES_AND_DEMO', 638503.734939759),
 ('HOUSE_AND_HOME', 1331540.5616438356),
 ('HEALTH_AND_FITNESS', 4188821.9853479853),
 ('GAME', 15588015.603248259),
 ('FOOD_AND_DRINK', 1924897.7363636363),
 ('FINANCE', 1387692.475609756),
 ('FAMILY', 3695641.8198090694),
 ('EVENTS', 253542.22222222222),
 ('ENTERTAINMENT', 11640705.88235294),
 ('EDUCATION', 1833495.145631068),
 ('DATING', 854028.8303030303),
 ('COMMUNICATION', 38456119.16724

In [45]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we removed all the communication apps that have over 100 million installs the average would be reduced roughly ten times:

In [46]:
under_100_m = []
for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] =='COMMUNICATION')and(float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
sum(under_100_m)/len(under_100_m)
    

3603485.3884615386

We see the aame pattern for the video players category, which is the runner-up with 24727872 install.The market is dominated by apps like Youtube, google play Movies& TV , or MX players. The pattern is repeated for social apps(where we have giants like facebook),photography apps,or productivity apps.

Again, the main concern is that these app genres might seem more popular than they really are.Moreover, these niches seem to be dominated by a few giants who are hard to compete againsr.

The game genre seems pretty popular but previously we found out this part of the mark sees a bit saturated , so we'd like to come up with some new idea if possible

The books and reference genre looks faairly popular aw well with an average number of installs of 8767811. It's intersting to explore this in more depth,since we found this genre has some potentual to work well on the app store, and our aim is to recommend an app genre thay show potentual for being profitable on both google and apple store.
 

In [48]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0],':',app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and refereence genre includes a variety of apps :software for processomh and reading ebooks,various collections of libraries,dictionaries, tutorials on programming or languages,etc.It seems there's still a small number of extremely popular apps that skew the averages:


In [49]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However , it look like there are only few very popilar apps , so this marker still shows potential.
Les't try to get some app based on the kind od apps that are somewhere in the middle terms of popilarity ()

In [51]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dicrionaries, it's properly not a good idea to build a smilar apps since there'll be some significant competition.

We also notice ther are quite a few apps built around the book Quran which suggests that building the app around a popular book may be profitable. It seems thay taking a popular book (perhaps more recent book ) and turning it into an app could be profitable for both google and app store markets.

However, it looks like the market is already full of libraraie, so we need to add some special features bedsides the raw version of the book. This might inclde daily quotes from the book and audio version of the book,quizzes on the book a forum whew poeple can discuss the book,etc.

# Conclusion 

In this project, we analyzed daya about the app store and google play mobile apps whtih the goal of recommending an profile that can be profurable for both markets.

We concluded thay taking a popular book (perhaps a more recent book) and turnimh it into an app could be profitable for both the google play and app stroe markets.The markets are already full of libraries, se we need to add some special features besides rhe raw version of the book.This might include daily quotes from the book,an audio versionof the book,quizzes on the book,a forum where people can discuss the book,etc.
